### PR TITLE
build: add MacOS and Windows to CI for landed commits

### PR DESCRIPTION
**Pull request by gregmagolan**

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -47,19 +47,44 @@ jobs:
...
+    steps:
+      - uses: actions/checkout@v3
+      - id: linux
+        run: echo "os=ubuntu-latest" >> $GITHUB_OUTPUT
+      - id: macos
+        run: echo "os=macos-latest" >> $GITHUB_OUTPUT
+        # Only run on main branch (not PRs) to minimize macOS minutes (billed at 10X)
+        # https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions#included-storage-and-minutes
+        # TODO: DNL
+        # if: ${{ github.ref == 'refs/heads/main' }}
```
Comment (alexeagle) [inline .github/workflows/ci.yaml:62]: nit: you could use a magic branch name that turns this on, then push your PR from there, then you don't need to comment/uncomment a line for testing

Comment (gregmagolan) [reply .github/workflows/ci.yaml]: that's a good idea; will use that next time I'm working on these

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -116,19 +177,30 @@ jobs:
...
         env:
           # Bazelisk will download bazel to here
           XDG_CACHE_HOME: ~/.cache/bazel-repo
 
+      - name: bazel test //... (Windows)
+        if: ${{ matrix.os == 'windows-latest' }}
+        working-directory: ${{ matrix.folder }}
+        run: |
+          $BAZEL_MAJOR_VERSION = "${{ matrix.bazelversion }}".substring(0, 1)
+          bazel --bazelrc=${{ github.workspace }}/.aspect/bazelrc/ci.bazelrc --bazelrc=${{ github.workspace }}/.github/workflows/ci.bazelrc --bazelrc=${{ github.workspace }}/.aspect/bazelrc/bazel$BAZEL_MAJOR_VERSION.bazelrc --bazelrc=.bazelrc test --config=${{ matrix.config }} //...
```
Comment (kormide) [inline .github/workflows/ci.yaml:198]: nit: Can you split this one onto multiple lines like we do in `bazel test //...` so it's easier to parse?

Comment (kormide) [reply .github/workflows/ci.yaml:198]: Perhaps also a comment clarifying why we need a different step to build on windows.

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -116,19 +177,30 @@ jobs:
...
+        env:
+          # Bazelisk will download bazel to here
+          XDG_CACHE_HOME: ~/.cache/bazel-repo
+
       - name: integration tests
-        if: ${{ matrix.folder == '.' }}
+        if: ${{ matrix.folder == '.' && matrix.os != 'windows-latest' }}
```
Comment (kormide) [inline .github/workflows/ci.yaml:204]: Why skip integration tests on Windows?

Comment (gregmagolan) [reply .github/workflows/ci.yaml:204]: They are bash scripts and Windows is running powershell

**Review by kormide — APPROVED** [review]

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (gregmagolan) [reply .github/workflows/ci.yaml:198]: Yes. It looks like you can use backticks to make multiline commands on Powershell

File: .github/workflows/ci.yaml (hunk 3 above)
Comment (gregmagolan) [reply .github/workflows/ci.yaml:210]: Add comment

**Pull request merged**